sw/airborne/math/wls/wls_alloc.c

```diff
@@ -45,22 +44,26 @@
 #include "math/qr_solve/qr_solve.h"
 #include "math/qr_solve/r8lib_min.h"
 
-void print_final_values(int n_u, int n_v, float* u, float** B, float* v, float* umin, float* umax);
-void print_in_and_outputs(int n_c, int n_free, float** A_free_ptr, float* d, float* p_free);
-
 // provide loop feedback
+#ifndef WLS_VERBOSE
 #define WLS_VERBOSE FALSE
+#endif
+
+#if WLS_VERBOSE
+#include <stdio.h>
+static void print_final_values(int n_u, int n_v, float* u, float** B, float* v, float* umin, float* umax);
+static void print_in_and_outputs(int n_c, int n_free, float** A_free_ptr, float* d, float* p_free);
+#endif
 
-// Problem size needs to be predefined to avoid having to use VLAs
-#ifndef CA_N_V
-#error CA_N_V needs to be defined!
+#ifndef WLS_N_U
```

Comment:
I think we should keep the comment about VLAs, for the next time we forget why we did it this way.

I think removing the error if N_V or N_U is not defined will lead to problems, someone can forget and try to fly an octocopter with this.

Also, I think it is now a little too easy to make a mistake with N_U or N_V, because you have to check yourself (a) weather you use both stab and guidance WLS (b) which one is larger. Can we not force the user to explicitly define both, and then we pick the largest one?

Reply:
I added size checks in stabilization and in guidance to be sure the matrix is big enough.